src/buttons/src/TextDropdownButton.tsx

```diff
@@ -1,56 +1,61 @@
-import React, { PureComponent } from 'react'
+import React, { PureComponent, ReactElement } from 'react'
 import PropTypes from 'prop-types'
 import Box from 'ui-box'
 import { Text } from '../../typography'
-import { Icon } from '../../icon'
+import { Icon, IconName } from '../../icon'
 import { Spinner } from '../../spinner'
-import { withTheme } from '../../theme'
+import { withTheme, Theme } from '../../theme'
 
-class TextDropdownButton extends PureComponent<
-  any & React.ComponentProps<typeof Box>
-> {
+interface TextDropdownButtonProps extends React.ComponentProps<typeof Box> {
```

Comment:
Hm, I think this should technically extend `TextProps` because it passes `...props` to `<Text />` not `<Box />`

Reply:
Ah, true.